### PR TITLE
chore(core): add eth address validator

### PR DESCRIPTION
**Pull request by JoshuaC817**
#### What this PR does / why we need it:
- Validate ETH address using ethers package
- Decode address hash for DFC and ETH

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

**Review by fullstackninja864 — CHANGES_REQUESTED** [review]

File: packages/walletkit-core/src/api/addressValidator.ts [hunk 1]
```diff
@@ -9,3 +10,6 @@ export const validateAddress = (
   const isValid = decodedAddress !== undefined;
   return isValid;
 };
+
+export const validateEthAddress = (address: string): boolean =>
```
[the remark this answers is not in the document]
Comment (pierregee) [reply packages/walletkit-core/src/api/addressValidator.ts:14]: No use case for it on DeFiScan, we can remove

File: packages/walletkit-core/src/api/addressValidator.unit.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
 };
 
+const ETH_ADDRESS = {
+  valid: "0x8ba1f109551bD432803012645Ac136ddd64DBA72",
+  validICAP: "XE65GB6LDNXYOFTX0NSV3FUWKOWIXAMJK36",
+  invalid: "0x8Ba1f109551bD432803012645Ac136ddd64DBa72",
+  invalidICAP: "0x8Ba1f109551bD432803012645Ac136ddd64DBA72",
+  random: "testing invalid eth address",
+};
+
```
Comment (pierregee) [inline packages/walletkit-core/src/api/addressValidator.unit.ts:16]: Once `validateEthAddress`, let's move these tests on `getDecodedAddress`

**Review by pierregee — CHANGES_REQUESTED** [review]

File: packages/walletkit-core/src/api/addressValidator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoshuaC817) [reply packages/walletkit-core/src/api/addressValidator.ts]: Done removed

File: packages/walletkit-core/src/api/addressValidator.unit.ts (hunk 2 above)
Comment (JoshuaC817) [reply packages/walletkit-core/src/api/addressValidator.unit.ts]: currently PushData in Script would only accept hex, so the addresses above could not be used in the test cases

**Comment by fullstackninja864** [comment]
Do we need to export `getDecodedAddress` in `packages/walletkit-core/src/api/index.ts` ?

**Review by fullstackninja864 — APPROVED** [review]

**Pull request merged**